src/networkaddress.rs

```diff
@@ -59,17 +59,22 @@ impl NetworkAddress for SocketAddrV4 {
                 >= std::mem::align_of::<libc::sockaddr_in>()
         );
 
-        let mut result = zeroed_sockaddr_storage();
-        // Safety: the above assertions guarantee that alignment and size are correct.
-        // the resulting reference won't outlast the function, and result lives the entire
-        // duration of the function
-        let out = unsafe { &mut (*(&mut result as *mut _ as *mut libc::sockaddr_in)) };
-        out.sin_family = libc::AF_INET as _;
-        out.sin_port = u16::from_ne_bytes(self.port().to_be_bytes());
-        out.sin_addr = libc::in_addr {
-            s_addr: u32::from_ne_bytes(self.ip().octets()),
+        let sockaddr = libc::sockaddr_in {
+            sin_family: libc::AF_INET as _,
+            sin_port: u16::from_ne_bytes(self.port().to_be_bytes()),
+            sin_addr: libc::in_addr {
+                s_addr: u32::from_ne_bytes(self.ip().octets()),
+            },
+            sin_zero: [0; 8],
         };
 
+        // # Safety
+        //
+        // The assertions above guarantee the validity of the cast. The asserts and the fact that
+        // result is allocated on the stack guarantee the validity of the write
+        let mut result = zeroed_sockaddr_storage();
+        unsafe { std::ptr::write(&mut result as *mut _ as *mut libc::sockaddr_in, sockaddr) };
```

Comment:
don't even bother with making a reference and the potential safety implications that has; just write to a pointer

Reply:
see above